### PR TITLE
Drop Node 12 and Node 14 (#5012)Co-authored-by: Gytis Cepkauskas <GytisCepk@users.noreply.github.com> Co-authored-by: Paul Connelly <22944042+pmconne@users.noreply.github.com> Co-authored-by: Ben <ben-polinsky@users.noreply.github.com>

### PR DESCRIPTION
**Pull request by GytisCepk**
### Node 12 and Node 14 have reached their End-of-Life and shouldn't be supported in 4.0.

Changes:
- Change ranges to support only Node 16 and Node 18.
- Update pipelines to run only on newer Node versions.
- Update documentation.

Node 16 will be dropped later this month.

### CODE REVIEW

**Comment by pmconne** [comment]
You'll add a note to NextVerion.md in the forthcoming Node 16 PR?

**Comment by GytisCepk** [comment]
> You'll add a note to NextVersion.md in the forthcoming Node 16 PR?

Updated in case Node 16 doesn't get dropped for some reason.

**Review by pmconne — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

File: common/config/azure-pipelines/jobs/docs-build.yaml [hunk 1]
```diff
@@ -35,9 +35,9 @@ jobs:
         clean: true
 
       - task: NodeTool@0
-        displayName: Use Node 14
+        displayName: Use Node 18
```
[the remark this answers is not in the document]
Comment (ben-polinsky) [reply common/config/azure-pipelines/jobs/docs-build.yaml:38]: Yeah it definitely runs on node 18. Weird, no other docs build runs are failing with these errors.

[the remark this answers is not in the document]
Comment (ben-polinsky) [reply common/config/azure-pipelines/jobs/docs-build.yaml:38]: Oh just saw we were moving from 14 on the build. Let me see if I can figure out what's going on.

**Review by MichaelBelousov — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

File: common/config/azure-pipelines/jobs/docs-build.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ben-polinsky) [reply common/config/azure-pipelines/jobs/docs-build.yaml:38]: I've reproduced the same behavior locally (node v14 ok / node v16+ fails)...

[the remark this answers is not in the document]
Comment (aruniverse) [reply common/config/azure-pipelines/jobs/docs-build.yaml:38]: looks good now, thx @ben-polinsky

**Pull request merged**